src/definitions.jl

```diff
@@ -356,6 +356,25 @@ plan_irfft(x::AbstractArray{Complex{T}}, d::Integer, region; kws...) where {T} =
 Pre-plan an optimized inverse real-input FFT, similar to [`plan_rfft`](@ref)
 except for [`irfft`](@ref) and [`brfft`](@ref), respectively. The first
 three arguments have the same meaning as for [`irfft`](@ref).
+
+!!! note
+    In contrast to plans created by [`plan_ifft`](@ref), the application of
+    this plan via `mul!(A, P, Â)` will potentially **overwrite the input array** `Â`
+    in the case of multidimensional transforms! See example below
+
+# Examples
+
+```
+julia> begin
+           A = Matrix{Float64}(undef, 5, 3)
+           Â = randn(ComplexF64, 3, 3)
+           Â_orig = copy(Â)
+           P = plan_irfft(Â, size(A, 1))
+           mul!(A, P, Â)
+           Â == Â_orig
+       end
+false
```

Comment:
Not sure how you feel about these examples.
These are not doctested, because they would require (and very likely depend on) a specific implementation like FFTW.jl.

On the other hand, the documentation of FFTW simply refers to the documentation of AbstractFFTs for all methods that are part of AbstarctFFTs, so this is currently really the only place where such examples would fit in.

Reply:
My inclination is to ditch the examples.

We should also make it clear that the input array is overwritten with undefined values (not with the output data, which still goes in `A`).

Reply:
> My inclination is to ditch the examples.

dropped.

> make it clear that the input array is overwritten with undefined values

added